File: Leetcode/java/sol121.java

// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/submissions/1648417938/

class Solution {
    public int maxProfit(int[] prices) {

        int min_price = prices[0];
        int max_profit = 0;
        for (int i : prices) {
            if(min_price > i){
                min_price = i;
            }
            max_profit = max_profit < i - min_price ? i - min_price:max_profit;

        }
        return max_profit;
    }
}

File: Leetcode/typescript/sol121.ts

// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/description/


function maxProfit(prices: number[]): number {
    let minCost = prices[0];
    let maxProfit = 0;
    for(const price of prices){
        // minCost = Math.min(minCost, price);
        minCost = price < minCost ? price:minCost
        // maxProfit = Math.max(maxProfit, price - minCost);
        maxProfit = price - minCost > maxProfit ? price - minCost: maxProfit
    }
    return maxProfit;
};